src/overlays/actors/ovl_En_Wallmas/z_en_wallmas.h

```diff
@@ -8,10 +8,26 @@
 #define EN_WALLMAS_GET_SWITCH_FLAG(thisx) (((thisx)->params >> 8) & 0xFF)
 #define EN_WALLMAS_IS_FROZEN(thisx) ((thisx)->params & 0x80)
 
+/**
+ * This type determines under what conditions the Wallmaster will drop from the ceiling.
+ * - EN_WALLMAS_TYPE_TIMER_ONLY: These Wallmasters don't check for anything to determine
+ *   whether or not to drop; they just set a 130-frame timer and drop once it reaches 0.
+ * - EN_WALLMAS_TYPE_PROXIMITY: These Wallmasters ensure that the player in within a certain
+ *   distance of their current position (as dictated by detectionRadius); once the player
+ *   enters that range, they set a 130-frame timer and drop once it reaches 0.
+ * - EN_WALLMAS_TYPE_FLAG: These Wallmasters wait for a certain switch flag to be activated,
+ *   as specified in their params. Once this switch is activated, they set an 81-frame timer
+ *   and drop once it reaches 0.
+ * 
+ * While in the "waiting to drop" state, all Wallmaster variations can be forced to reset the
+ * timer to 130 frames if the player enters certain states (e.g., if the player is no longer
+ * touching the ground). Additionally, EN_WALLMAS_TYPE_PROXIMITY-type Wallmasters will reset
+ * the timer to 130 frames if the player strays too far from the Wallmaster's current position.
+ */
 typedef enum {
```

Comment:
Ah, much clearer, thanks. Although I'd be fine with using `WALLMASTER` as the prefix since it's the same length.